Add ergodash shield

Eyenseo opened the pull request.

Nicell commented on app/boards/shields/ergodash/ergodash.keymap:
```diff
@@ -0,0 +1,95 @@
+ * .----------------------------------------------------------------------------------------------------------------------.
+ */
+                        bindings = <
+&kp  GRAVE    &kp  N1       &kp  N2       &kp  N3       &kp  N4       &kp  N5       &kp  LBKT              &kp  RBKT     &kp  N6       &kp  N7       &kp  N8       &kp  N9       &kp  N0       &kp  PRSC
+&kp  ESC      &kp  Q        &kp  W        &kp  F        &kp  P        &kp  G        &kp  MINUS             &kp  EQUAL    &kp  J        &kp  L        &kp  U        &kp  Y        &kp  SEMI     &kp  BSLH
+&kp  TAB      &kp  A        &kp  R        &kp  S        &kp  T        &kp  D        &kp  DEL               &kp  BSPC     &kp  H        &kp  N        &kp  E        &kp  I        &kp  O        &kp  SQT
+&kp  LSHFT    &kp  Z        &kp  X        &kp  C        &kp  V        &kp  B        &kp  END               &kp  HOME     &kp  K        &kp  M        &kp  COMMA    &kp  DOT      &kp  SLASH    &kp  RSHFT
+&kp  LCTRL    /**/ &none    &kp  PG_DN    &kp  LMETA    &kp  LALT     &kp  SPACE    &mo  LOWER             &mo  RAISE    &kp  RET      &kp  RALT     /**/ &none    &kp  PG_UP    &kp  INS      &kp  RCTRL
+                        >;
+                };
+                lower_layer {
+/* .----------------------------------------------------------------------------------------------------------------------.
+ * |  F11 |  F1  |  F2  |  F3  |  F4  |  F5  |      |                    |      |  F6  |  F7  |  F8  |  F9  |  F10 |  F12 |
+ * |------+------+------+------+------+------+------+--------------------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |      |      |                    |      |      |      |      |      |      |      |
+ * |------+------+------+------+------+------+------+--------------------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |      |      |                    |      |      |      |      |      |      |      |
+ * |------+------+------+------+------+------+---------------------------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |      |      |                    |      |      |      |      |      |      |      |
+ * |-------------+------+------+------+------+------+------+------+------+------+------+------+------+------+-------------|
+ * | REST | BOOT |      |      |      |      |      |      |      |      |      |      |      |      |      |      |      |
+ * .----------------------------------------------------------------------------------------------------------------------.
+ */
+                        bindings = <
+&kp  F11      &kp  F1       &kp  F2       &kp  F3       &kp  F4       &kp  F5       &kp  &none             &kp  &none    &kp  F6       &kp  F7       &kp  F8       &kp  F9       &kp  F10      &kp  F12
+/**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none             /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none
+/**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none             /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none
+/**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none             /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none
+&reset        &bootloader   /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none             /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none    /**/ &none
```
I appreciate the effort that's gone into adding these comments to line up `&none`, but I'm wondering if they might cause maintenance issues/be annoying for when people customize their keymap.

Eyenseo replied:
It wasn't really that much effort - removed.

Nicell reviewed (changes requested):
I have a couple comments to start, and then could you add this shield to the following files, please?

[`docs/docs/hardware.md`](https://github.com/zmkfirmware/zmk/blob/main/docs/docs/hardware.md): You should add the Ergodash to this list

[`docs/static/setup.ps1`](https://github.com/zmkfirmware/zmk/blob/main/docs/static/setup.ps1) and [`docs/static/setup.sh`](https://github.com/zmkfirmware/zmk/blob/main/docs/static/setup.sh): Ergodash should be added as a split option for both files.

Eyenseo commented:
I'll make another change later today. I'll change the Colemak layout to `querty` so it's easier for people to reuse.

Eyenseo commented:
@Nicell @petejohanson my last commit is somewhat funky. Is there an issue that tracks this behaviour? Apparently just references is not allowed in a keymap.

Daegalus commented:
Is there anything blocking this being merged? I just tested this on my Ergodash build and it works well. I redid the maps for myself but everything mapped correctly and it is working well.

Eyenseo commented:
@Daegalus :shrug:

I rebased onto main again, removed the hack to compile - that is apparently no longer necessary. Once reviewed I would squash the commits so it can be merged.

michelv commented:
Hi! Ergodash user here, I'll test once I have prepared my nice!nano's for the swap.

From the keymap file though, it looks like it's missing support for those who have more thumb keys (two spacebars per half for example). Or am I misreading the file?

Eyenseo commented:
@michelv This layout is for the second from the top [1] - I'm not sure how the other layouts differ. But I'm sure that that can be configured with defines :)

[1] https://raw.githubusercontent.com/omkbd/picture/master/ergodash-layout.png

Daegalus commented:
You have to remember that the ergodash only has 4 possible keys in the thumb cluster. Depending on your configuration a 2u key using the center slot will eat up 2 keys and will be set by the same key in the config. You have to play around with it a bit. But the layout is correct.

Daegalus commented:
Any updates on getting this merged?

petejohanson commented:
Thank you, contributor, for your patience with how long review and merge of boards/shields has taken!

There are three recent refactors/changes to boards and shields that require some attention, and then we can finally get this PR merged!

1. Hardware Metadata
1. Pro Micro shield DT naming changes
1. Split changes for BLE advertising

# Hardware Metadata

## The Problem

When first developing the process around contributing new shields/boards to ZMK, we failed to recognize that several key files (setup scripts, documentation page of supported hardware, and GH Action `build.yml` file) required changes, often in the same spot, for every PR. This resulted in immediate merge conflicts for every *other* PR after one was merged, which is a  headache for contributors.

## The Fix

By adding discrete metadata files that are located with the boards/shields in question, and using that metadata to generate setup scripts, website hardware list, etc., users can contributing new hardware descriptions without the need to change the same files that other contributors are changing.

## Next Steps

First, refer to https://zmk.dev/docs/development/hardware-metadata-files to familiarize yourself with the new metadata file format.

Next, you have two options for fixing up your PR:

1. If comfortable with `git rebase`, perform an interactive rebase and revert any changes to `build.yml`, `hardware.md`, or the setup scripts `setup.sh`/`setup.ps1`, and then add the new metadata YAML file. Then force push your branch. Or,
2. Create a new branch from an up-to-date `main`, copy in the files for your new hardware, add the metadata file, then commit and push the new branch. Then, edit your open PR to point to your new branch.

# Pro Micro shield DT naming changes

In #876, we have simplified the DT naming for the "nexus node" we expose for pro-micro compatible boards, deprecating the use of `pro_micro_a`, and renaming `pro_micro_d` node to simply `pro_micro`. For pro-micro boards and shields, you'll need to adjust your DT to use the proper names.

Please see https://zmk.dev/docs/development/new-shield#shield-overlays for the updated docs on this.

# Split Shield Advertising Changes

In addition, if this is a split PR, please see #658 where we have changed our conventions to *remove* the the name from the right sides, to prevent users attempting to pair with them and causing split sync issues. This also includes removing the " Left" suffix from the naming on the left side. See the changes in that PR for examples of what to change with your split shield.

# Getting Help

If you have any questions about any of these changes, please comment here and tag @zmkfirmware/boards-shields or ask in the `#boards-shields` Discord channel.

Daegalus commented:
@petejohanson Looks like the PR owner has updated the repo with your requested changes, is there anything else that needs to be done?
